### DISCUSSION TITLE
FieldArray not cleared and default values not used when resetting the form

### DISCUSSION THREAD

**Discussion by ergunp** [react-hook-form/react-hook-form#3030] [Q&A] [answered]
Hi,

I am using a nested field array which when a user appends a new item to the array after the user has clicked on reset button the form should remove that added item and reset to default values. Here is a sandbox example

https://codesandbox.io/s/react-hook-form-usefieldarray-nested-arrays-forked-69mxu?file=/src/golden-build-create-order.jsx

when the user clicks on the add "interConnection" button and then clicks on reset, I'd expect it to remove that item but it is not the case. 

However when a user clicks on the "Add patching" and clicks on the reset form, it does remove that item and resolve to the default value. 

I am not sure why its not working for the interConnection nested array. 

Using: 6.8.6

Any help would be appreciated.

**Comment by bluebill1049** [comment, marked as answer]
hey @ergunp, any chance you can reproduce this in an isolated simpler csb?

**Reply by ergunp** [reply]
Hey @bluebill1049 I have made the code much simpler to understand and it it is reproducible, please take a look https://codesandbox.io/s/react-hook-form-usefieldarray-nested-arrays-forked-69mxu?file=/src/patching-form.jsx 

if you click on the add interconnection button then if you click on the reset button the added interconnection is not removed.

thanks

**Reply by bluebill1049** [reply]
you probably want to supply a value when you are using with Controller: `reset({});`

**Reply by bluebill1049** [reply]
<img width="880" alt="Screen Shot 2020-09-30 at 7 48 32 pm" src="https://user-images.githubusercontent.com/10513364/94670466-edb38200-0355-11eb-8b84-6bf0371e9279.png">

**Reply by ergunp** [reply]
I have added  `reset(defaultValues);` but still having the same issues

**Reply by bluebill1049** [reply]
i have removed the setValue in front of reset, it seems to do things more correctly. however, there are more issue with your codesandbox

eg: key is not using from the field and probably more issue within your codesandbox.
<img width="467" alt="Screen Shot 2020-09-30 at 8 00 52 pm" src="https://user-images.githubusercontent.com/10513364/94671782-a75f2280-0357-11eb-8823-bb3a8720ea79.png">


here is the working version which i just did with on top of our example.
https://codesandbox.io/s/react-hook-form-usefieldarray-nested-arrays-x7btr?file=/src/index.js

**Reply by ergunp** [reply]
adding `React.Fragment key={item.id}` seems to have fixed the issue, thank for the help, much appreciated.
